### PR TITLE
Remove unecessary conversion functions between `Felt` & `BigUint`/`BigInt`

### LINKED ISSUE #1495: Refactor biguint_to_felt and bigint_to_felt functions
We should delete the functions `felt_to_biguint`, `felt_to_bigint`, `biguint_to_felt` and `bigint_to_felt`
https://github.com/lambdaclass/cairo-vm/blob/55eaaa7be3778b84871e24d1f87e4ae6f62184dd/vm/src/utils.rs#L67
And use the BigUint and BigInts conversion to felt, provided by the `starknet-types-core` crate

### PR DESCRIPTION
**Pull request by fmoletta**
Closes #1495

### CODE REVIEW

File: vm/src/hint_processor/builtin_hint_processor/ec_utils.rs [hunk 1]
```diff
@@ -144,10 +144,10 @@ pub fn recover_y_hint(
     let p_x = get_integer_from_var_name("x", vm, ids_data, ap_tracking)?.into_owned();
     let p_addr = get_relocatable_from_var_name("p", vm, ids_data, ap_tracking)?;
     vm.insert_value(p_addr, p_x)?;
-    let p_y = biguint_to_felt(
-        &recover_y(&felt_to_biguint(p_x))
+    let p_y = Felt252::from(
+        &recover_y(&p_x.to_biguint())
```
Comment (Oppen) [inline vm/src/hint_processor/builtin_hint_processor/ec_utils.rs:148]: I see some redundant conversion here. Maybe `recover_y` could simply return the `Felt252` instead.

**Review by Oppen — COMMENTED** [review]
Nice cleanup. There's some slowdown that may be worth investigating.

File: vm/src/hint_processor/builtin_hint_processor/ec_utils.rs (hunk 1 above)
Comment (fmoletta) [reply vm/src/hint_processor/builtin_hint_processor/ec_utils.rs:148]: `recover_y` is also used by other functions that use BigUint and don't need to convert  from Felt to use it, so it is better to leave the conversion outside and let the function just handle BigUint

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 #### Upcoming Changes
 
+* feat: Remove unecessary conversion functions between `Felt` & `BigUint`/`BigInt` [#1562](https://github.com/lambdaclass/cairo-vm/pull/1562)
+  * Remove the following functions:
+    * felt_from_biguint
+    * felt_from_bigint
+    * felt_to_biguint
+    * felt_to_bigint
```
Comment (Oppen) [inline CHANGELOG.md:10]: These functions were public, so we need to report the change as a breaking one.

**Review by Oppen — DISMISSED** [review]

**Review by pefontana — APPROVED** [review]

**Review by pefontana — DISMISSED** [review]

**Review by juanbono — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

**Pull request merged**